src/App.js

```diff
@@ -121,40 +103,48 @@ class App extends Component {
 
   render() {
     return (
-      <main className="App">
-      {this.state.user ?
-        <Header 
-          buttonDisplay={this.logOut} 
-          user={this.state.user}
-          buttonText='Log Out'
-          title={`Welcome ${this.state.user.name}`}
-          /> :
-        <Header 
-          buttonDisplay={this.showLoginPage}
-          buttonText='Log In'
-          title='Welcome to Rancid Tomatillos'
-          />
-      }
-        {this.state.login &&
-          <Login
-            fetchUserData={this.fetchUserData} 
-            closeLoginPage={this.closeLoginPage}/>}
-        {this.state.homepage &&
-          <section className="home-page">
-            <CardContainer 
-              movies={this.state.movies} renderSpecificMovie={this.renderSpecificMovie}/>
-          </section>
-        }
-        {this.state.selectedMovie &&
-            <Movie 
-              movie={this.state.selectedMovie}
-              closeMovieDetail={this.closeMovieDetail}
+      <BrowserRouter>
+        <main className="App">
+        {this.state.user ?
+          <Header 
+            buttonDisplay={this.logOut} 
+            user={this.state.user}
+            buttonText='Log Out'
+            title={`Welcome ${this.state.user.name}`}
+            /> :
+          <Header 
+            buttonDisplay={this.showLoginPage}
+            buttonText='Log In'
+            title='Welcome to Rancid Tomatillos'
             />
         }
-        {this.state.error &&
-          <ErrorPage errorCode={this.state.error}/>
-        }
-      </main>
+          <Switch>
+            <Route path="/" exact render={() => 
```

Comment:
This is a good place for a redirect! If someone visits your home page and they are not logged in, redirect them to the Login page

Reply:
Ohh redirect, i'll have to look into that and add that implementation, I was having a problem getting it to react the way I wanted and I think that may be the solution I was looking for. Thank you!